python/pyproject.toml

```diff
@@ -107,7 +107,7 @@ force_grid_wrap = 4
 all = true
 
 [tool.mypy]
-no_implicit_optional = true
```

Comment:
@LuigiCerone I think we want to keep this setting to avoid bugs creeping in

Reply:
@Fokko Yep, you're right. I'll revert that thanks :)